.travis.yml

```diff
@@ -17,85 +12,76 @@ before_install:
 install:
   -
 
-env:
-  -JDK=Java8
-  -JDK=Java9
-  -JDK=Java10
-  -JDK=Java11-EA
-  -JDK=Java12-EA
-
 matrix:
   fast_finish: true
   include:
     - jdk: oraclejdk8
       env:
         - DESC="acceptance tests"
-        - JDK=Java8
         - CMD="mvn install --projects acceptance-tests --also-make --activate-profiles all"
 
     - jdk: oraclejdk8
       env:
         - DESC="findbugs"
-        - JDK=Java8
         - CMD="mvn install findbugs:check --projects '!scala-unit-tests,!jmh-scala-tests,!jmh-tests,!jcstress-tests,!p2-repository' --activate-profiles all -DskipTests=true"
+
     - jdk: oraclejdk8
       env:
         - DESC="checkstyle"
-        - JDK=Java8
         - CMD="mvn install checkstyle:check --activate-profiles all -DskipTests=true"
 
     - jdk: oraclejdk8
       env:
         - DESC="unit tests"
-        - JDK=Java8
         - CMD="mvn install"
 
     - jdk: oraclejdk8
       env:
         - DESC="compile performance-tests"
-        - JDK=Java8
         - CMD="mvn install --projects performance-tests --also-make --activate-profiles all -DskipTests=true"
 
     - jdk: oraclejdk9
       env:
-        - DESC="unit tests Java 9"
-        - JDK=Java9
+        - DESC="unit tests oraclejdk9"
         - CMD="mvn install"
 
-    - jdk: oraclejdk8
+    - jdk: openjdk10
+      env:
+      - DESC="unit tests openjdk10"
+      - CMD="mvn install"
+
+    - jdk: oraclejdk10
+      env:
+      - DESC="unit tests oraclejdk10"
+      - CMD="mvn install"
+
+    - jdk: openjdk11
       env:
-        - DESC="Java 10 regression"
-        - JDK=Java10
-        - CMD=""
+      - DESC="unit tests openjdk11"
+      - CMD="mvn install"
 
     - jdk: oraclejdk8
       env:
-        - DESC="Java 11 regression"
+        - DESC="unit tests Java11-EA"
         - JDK=Java11-EA
-        - CMD=""
+        - CMD="./.travis-command-ea-builds.sh"
 
     - jdk: oraclejdk8
       env:
-        - DESC="Java 12 regression"
+        - DESC="unit tests Java12-EA"
         - JDK=Java12-EA
-        - CMD=""
+        - CMD="./.travis-command-ea-builds.sh"
 
   allow_failures:
   - jdk: oraclejdk8
     env:
-      - DESC="Java 11 regression"
       - JDK=Java11-EA
-      - CMD=""
 
   - jdk: oraclejdk8
     env:
-      - DESC="Java 12 regression"
       - JDK=Java12-EA
-      - CMD=""
```

Comment:
To allow failures, I believe only the jdk and JDK (environment variable) are required.

Reply:
From the docs to allow failures the "allow_failures" block has to be exactly same as the matrix: https://docs.travis-ci.com/user/customizing-the-build/#rows-that-are-allowed-to-fail
It was broken before, but after #607 was merged you can see the build matrix changed on Travis to show the builds which allowed failures.
If you remove the cron condition, you should be able to see them in a different matrix labeled "Allowed Failures" that will tell us if this works or not.

Reply:
I can double check. I had tried it with an earlier commit, before adding the `if:` condition and I saw it work without DESC.